src/aurora/ltrfile.cpp

```diff
@@ -0,0 +1,117 @@
+//
+// Created by patrick on 02.05.18.
+//
+
+#include "src/common/readstream.h"
+#include "src/common/strutil.h"
+
+#include "src/aurora/ltrfile.h"
+#include "src/aurora/resman.h"
+
+static const uint32 kLTRID  = MKTAG('L', 'T', 'R', ' ');
+static const uint32 kVersion10 = MKTAG('V', '1', '.', '0');
+
+static const char* kLetters = "abcdefghijklmnopqrstuvwxyz'-";
+
+namespace Aurora {
+
+LTRFile::LTRFile(Common::UString name) {
+	Common::ScopedPtr<Common::SeekableReadStream> stream(ResMan.getResource(name, kFileTypeLTR));
+
+	readHeader(*stream);
+
+	if (_id != kLTRID)
+		throw Common::Exception("Not a LTR file (%s)", Common::debugTag(_id).c_str());
+
+	if (_version != kVersion10)
+		throw Common::Exception("Unsupported LTR file version %s", Common::debugTag(_version).c_str());
+
+	byte letterCount = stream->readByte();
+	if (letterCount != 28)
```

Comment:
There are files in the games data files that have only 26 letters allocated, `names.ltr`. Are those completely unused?

Otherwise, it might be useful adding support for other letter counts than 28. Maybe pop in a TODO comment here to.

Reply:
Which game exactly uses the names.ltr? I would assume that this simply means, that it is using the standard alphabet without ' and -.

Reply:
Both NWN and NWN2 have a names.ltr with 26 letters in their data files. Dunno whether they're used, though

Reply:
I looked into nwn, and it seems that the name generation is already using species dependent ltr files. But i have also added for the sake of completeness a second mode, where the standard alphabet without ' and - is used.

Reply:
A second mode? Can't you just make the number of letters completely variable? As far as I understood, the letter set is fixed, and the less letters, the more letters from the end are left out, no?

Reply:
Well, I cannot say for sure, that they do not use other special characters in more than 26 letter alphabets.